client/galaxy/scripts/layout/router.js

```diff
@@ -31,26 +28,8 @@ var Router = Backbone.Router.extend({
         var queryObj = QUERY_STRING.parse(args.pop());
         args.push(queryObj);
         if (callback) {
-            if (this.authenticate(args, name)) {
-                callback.apply(this, args);
-            } else {
-                this.access_denied();
-            }
+            callback.apply(this, args);
         }
-    },
-
-    authenticate: function(args, name) {
```

Comment:
This isn't dead code -- it's a default passthrough.  The router is extended elsewhere, and authenticate is given functionality, which then inter-operates with the other methods here.

Reply:
Good point, do you know if this is meant as a client validation of requests (since we obviously have to have proper checks on the backend anyways)?

Reply:
Yes I do, and yes it is.  It's faster and more user-friendly to validate this stuff on the client-side when we can.

Reply:
As a fan of thin clients I disagree here but will respect that.

Reply:
(I removed the second commit form the PR)

Reply:
Thanks for stripping the second commit.  I think it's pretty widely accepted that clientside validation provides a significant UX improvement.